### PR TITLE
* Minor changes for succesful builds against .NET3.5

### PR DESCRIPTION
**Pull request by NPatch**
Trying to use this solution in Unity 5.6 which is still using Mono 2.0. Couldn't find an old dll so applying the minimum of changes to pass the build.

### CODE REVIEW

**Comment by NPatch** [comment]
By the way the actual changes in TemplatingEngine.cs are:
`
diff --git a/Mono.TextTemplating/Mono.TextTemplating/TemplatingEngine.cs b/Mono.TextTemplating/Mono.TextTemplating/TemplatingEngine.cs
index c73b510..150d33e 100644
--- a/Mono.TextTemplating/Mono.TextTemplating/TemplatingEngine.cs
+++ b/Mono.TextTemplating/Mono.TextTemplating/TemplatingEngine.cs
@@ -147,7 +147,7 @@ namespace Mono.TextTemplating
 				var obj = domain.CreateInstanceFromAndUnwrap (type.Assembly.Location, type.FullName, false,
 					BindingFlags.Default, null,
 					new object[] { host, results, templateClassFullName, settings.Culture, references.ToArray () },
-					null, null);
+					null, null,null);
 				return (CompiledTemplate)obj;
 			}
 			return new CompiledTemplate (host, results, templateClassFullName, settings.Culture, references.ToArray ());
@@ -167,7 +167,7 @@ namespace Mono.TextTemplating
 			
 			if (settings.Debug)
 				pars.TempFiles.KeepFiles = true;
-			if (string.IsNullOrWhiteSpace (pars.CompilerOptions))
+			if (StringUtil.IsNullOrWhiteSpace (pars.CompilerOptions))
 				pars.CompilerOptions = "/noconfig";
 			else if (!pars.CompilerOptions.Contains ("/noconfig"))
 				pars.CompilerOptions = "/noconfig " + pars.CompilerOptions;

`

**Comment by NPatch** [comment]
Also since I didn't want to mess with the .NET4+ compatibility in regards to IsEmptyOrWhitespace, I just created a totally different function for it. If you see a better way to resolve this ,please do so.

File: Mono.TextTemplating/Mono.TextTemplating/StringUtil.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.IO;
+
+namespace Mono.TextTemplating
+{
+	static class StringUtil
+	{
+		public static bool IsNullOrWhiteSpace(string value)
```
Comment (NN---) [inline Mono.TextTemplating/Mono.TextTemplating/StringUtil.cs]: Formatting issues.
Make either all tabs or spaces.

**Comment by NPatch** [comment]
@NN--- Fixed the tab-space issue. I think that was a VSCode blunder. By default Code uses spaces for tabs,which I didn't know. I wanted to quickly edit something without opening VS for it and used Code which probably changed some of the tabs.

**Comment by mhutch** [comment]
This has now been fixed by https://github.com/mono/t4/pull/24. Sorry for the delay!

**Pull request closed without merging**